include/xtensor/xgrid_iterator.hpp

```diff
@@ -0,0 +1,139 @@
+/***************************************************************************
+* Copyright (c) Johan Mabille, Sylvain Corlay and Wolf Vollprecht          *
+* Copyright (c) QuantStack                                                 *
+*                                                                          *
+* Distributed under the terms of the BSD 3-Clause License.                 *
+*                                                                          *
+* The full license is in the file LICENSE, distributed with this software. *
+****************************************************************************/
+
+#ifndef XTENSOR_XGRID_ITERATOR
+#define XTENSOR_XGRID_ITERATOR
+
+#include "xtl/xsequence.hpp"
+
+
+#include "xstrided_view.hpp"
+
+namespace xt
+{
+
+    template<class S>
+    class xsubgrid_iterator
```

Comment:
Nitpicking: I think `xmultiindex_iterator` or `xgridindex_iterator` would be more explicit about what it does.